collections/_posts/2023-07-06-Mean_teachers_are_better_role_models.md

```diff
@@ -0,0 +1,202 @@
+
+&rarr; focus on **regularization methods** in **semi-supervised learning**
+
+# Method 
+
+## Applying noise and Ensembling method
+
+### **Noise regularization**
+- Add noise to the input or intermediate representations of a model
+- Helps learning abstract invariances
+- **Pushes decision boundaries away from labeled data points**  
+
+Limitations of noise regularization :
+- classification cost is **undefined for unlabeled examples** &rarr; not helpful in semi-supervised learning.
+
+&rarr; The $$\Gamma$$ model to overcome this
+
+### **The $$\Gamma$$ model[^1]**
+- Evaluates each data point with and without noise, and then applies a consistency cost between the two predictions. 
+- Student teacher models where the **teacher generates targets**, which are then **used by itself as a student for learning**
+
+Limitations of $$\Gamma$$ model :
+- the model itself generates targets, which may be incorrect
+- the cost of inconsistency outweighs that of misclassification, preventing the learning of new information
+- can lead to a **confirmation bias**
+
+&rarr; Confirmation bias can be mitigated by improving the targets. One way is illustrated with the $$\Pi$$ model, selecting a different teacher model than the student model
+
+> **NOTE :** The authors mention another way "to choose the perturbation
+of the representations carefully instead of barely applying additive or multiplicative noise" wich is the subject of another paper by Miyato _et al._ (2017)[^2] that will be refered as VAT+EntMin latter. 
+
+### **The $$\Pi$$ model[^3]**
+- **noise** is added to the model **during inference**
+- results in a "noisy teacher" that can provide **more accurate targets**. 
+
+&rarr; The $$\Pi$$ model can be further improved by Temporal Ensembling (TE)
+
+### **Temporal Ensembling (TE)**
+- **exponential moving average (EMA)** prediction for each training example is formed by combining the **current version of the model's predictions** with the **predictions made by earlier versions of the model** that evaluated the same example. 
+- improves the quality of the prediction 
+- better results when those predictions are used as the teacher predictions
+
+Limitation of Temporal ensembling :
+- Each **target** (prediction) is **updated only once per training epoch**.
+- The learned information is **incorporated** into the training process at a relatively **slow pace**.
+- The **larger the dataset**, the **longer it takes** for the updates to span the entire dataset.
+
+&rarr; The authors propose the Mean Teacher model that **averages model weights**, instead of predictions like TE, to improve performances of semi-supervised model.
+
+![](/collections/images/mean_teachers/binary_classification_2.jpg)
+
+A sketch of a binary classification task with two labeled examples (large black dots) and one unlabeled example, demonstrating how the choice of the unlabeled target (blue circle) affects the fitted function (gray curve).
```

Comment:
To make things even more clear, you could add the name of the models that correspond to each figure. For example (d) corresponds to $\Pi$.